directory/datadog_checks/directory/data/conf.yaml.example

```diff
@@ -24,7 +24,8 @@ instances:
 
     ## @param filetagname - string - optional - default: <FILE_NAME>
     ## The name of the key for the tag used for each file, the value is the filename.
-    ## The resulting tag is "<filetagname>:<filename>".
+    ## The resulting tag is "<filetagname>:<filename>". Only used if filegauges is set to true and if the directory
+    ## contains 20 files or less.
```

Comment:
> if the directory contains 20 files or less.

Seems not completely true.

Looks like `"<filetagname>:<filename>"` tag is added for the first 20 files. But that's not exactly true either, since `directory_files` is incremented by batch of `directory_files += get_length(files)`, so the number of file metrics submitted with `"<filetagname>:<filename>"` might vary depending on the batch sizes.

Maybe instead of `if filegauges and directory_files <= 20:`, we could use a `submitted_files_count` that increases by 1 for each file processed. For example:

cc @hush-hush 

```python

        submitted_files = 0
        
        for root, dirs, files in walker:
            directory_files += get_length(files)

            for file_entry in files:
                [...]

                try:
                    file_stat = file_entry.stat()

                except OSError as ose:
                    [...]
                else:
                    if filegauges and submitted_files < 20:
                        [...]
                    else:
                        [...]
                submitted_files += 1

```

Reply:
For reference this line originated here: https://github.com/DataDog/dd-agent/commit/e9ccecbed1a568a099e20bbe5e4465a660366937

I think the idea was to be able to monitor specific files size without creating a huge number of new context (the hard-coded limit is here to prevent an explosion of context if the check is misconfigured).

It's true that right now the limit is "broken" since we increase by batch, but it prevent adding the filename as a tag for part of a directory. Since the order in which we handle files is not guarantee, adding the `<filename>` as a tag for 20 first file could result has different tags being sent at each run (and therefor an explosion in new context on the backend).

I think we should stick with the current implementation.